src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -427,194 +396,174 @@ createBlockHistoryTable =
+    insertargs =  [SText tablename, SInt (fromIntegral bh)]
+    tablename = Utf8 (toS $ asString name)
+    indexname = Utf8 (toS $ asString name <> "_bh")
+
+createVersionedTable :: Utf8 -> Utf8 -> Database -> IO ()
+createVersionedTable tablename indexname db = do
+    exec_ db createtablestmt
+    exec_ db indexcreationstmt
+  where
+    createtablestmt =
+      "CREATE TABLE IF NOT EXISTS["
+        <> tablename
+        <> "] (rowkey TEXT\
+             \, blockheight UNSIGNED BIGINT NOT NULL\
+             \, txid UNSIGNED BIGINT NOT NULL\
+             \, rowdata BLOB NOT NULL\
+             \, UNIQUE (blockheight, rowkey, txid));"
+    indexcreationstmt =
+        mconcat
+            ["CREATE INDEX IF NOT EXISTS ["
+            , indexname
+            , "] ON ["
+            , tablename
+            , "](rowkey, txid);"]
+
+handlePossibleRewind :: BlockHeight -> ParentHash -> BlockHandler SQLiteEnv TxId
+handlePossibleRewind bRestore hsh = do
     bCurrent <- getBCurrent
-    vCurrent <- getVCurrent
     checkHistoryInvariant
-
     case compare bRestore (bCurrent + 1) of
-        GT -> internalError "handleVersion: Block_Restore invariant violation!"
-        EQ -> newChildBlock bCurrent vCurrent
-        LT -> rewindBlock vCurrent
-
+        GT -> internalError "handlePossibleRewind: Block_Restore invariant violation!"
+        EQ -> newChildBlock bCurrent
+        LT -> rewindBlock bRestore
   where
     getBCurrent = do
-        r <- callDb "handleVersion" $ \db ->
+        r <- callDb "handlePossibleRewind" $ \db ->
              qry_ db "SELECT max(blockheight) AS current_block_height \
                      \FROM BlockHistory;" [RInt]
-        SInt bh <- liftIO $ expectSingleRowCol "handleVersion: (block):" r
+        SInt bh <- liftIO $ expectSingleRowCol "handlePossibleRewind: (block):" r
         return $! BlockHeight (fromIntegral bh)
-    getVCurrent = do
-        r <- callDb "handleVersion" $ \db ->
-             qry_ db "SELECT max(version) AS current_version \
-                     \FROM VersionHistory;" [RInt]
-        SInt version <- liftIO $
-                        expectSingleRowCol "handleVersion: (version):" r
-        return $! ReorgVersion version
 
     checkHistoryInvariant = do
-        -- enforce invariant that the history has (B_restore-1,H_parent).
-        historyInvariant <- callDb "handleVersion" $ \db -> do
+        -- enforce invariant that the history has
+        -- (B_restore-1,H_parent).
+        historyInvariant <- callDb "handlePossibleRewind" $ \db -> do
             qry db "SELECT COUNT(*) FROM BlockHistory WHERE \
-                   \blockheight = ? AND hash = ?;"
-                [ SInt $! fromIntegral $ pred bRestore
-                , SBlob (Data.Serialize.encode hsh) ]
-                [RInt]
-            >>= expectSingleRowCol "handleVersion: (historyInvariant):"
+                   \blockheight = ? and hash = ?;"
```

Comment:
revert this change, sql keywords are uppercase in all our queries

Reply:
done and done